src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs

```diff
@@ -561,39 +559,36 @@ private void SetColumnArgument(IChannel ch, object inputInstance, string argName
             }
         }
 
-        public static EntryPointNode Create(
-            IHostEnvironment env,
-            string entryPointName,
-            object arguments,
-            ComponentCatalog catalog,
-            RunContext context,
-            Dictionary<string, List<ParameterBinding>> inputBindingMap,
-            Dictionary<ParameterBinding, VariableBinding> inputMap,
-            Dictionary<string, string> outputMap,
-            bool checkpoint = false,
-            string stageId = "",
-            float cost = float.NaN)
+        public static EntryPointNode Create(IHostEnvironment env,
+          string entryPointName,
+          object arguments,
+          RunContext context,
+          Dictionary<string, List<ParameterBinding>> inputBindingMap,
+          Dictionary<ParameterBinding, VariableBinding> inputMap,
+          Dictionary<string, string> outputMap,
+          bool checkpoint = false,
+          string stageId = "",
+          float cost = float.NaN)
         {
             Contracts.CheckValue(env, nameof(env));
             env.CheckNonEmpty(entryPointName, nameof(entryPointName));
             env.CheckValue(arguments, nameof(arguments));
-            env.CheckValue(catalog, nameof(catalog));
             env.CheckValue(context, nameof(context));
             env.CheckValue(inputBindingMap, nameof(inputBindingMap));
             env.CheckValue(inputMap, nameof(inputMap));
             env.CheckValue(outputMap, nameof(outputMap));
             ComponentCatalog.EntryPointInfo info;
-            bool success = catalog.TryFindEntryPoint(entryPointName, out info);
+            bool success = env.ComponentCatalog.TryFindEntryPoint(entryPointName, out info);
             env.Assert(success);
 
-            var inputBuilder = new InputBuilder(env, info.InputType, catalog);
+            var inputBuilder = new InputBuilder(env, info.InputType, env.ComponentCatalog);
```

Comment:
Can you remove the ComponentCatalog param from this constructor too?

Reply:
InputBuilder's constructor takes IExceptionContext as first argument, not IHostEnvironment, and IExceptionContext doesn't have a ComponentCatalog property, so I think this one can't be changed.

Reply:
Ah, good point. Please ignore.